libraries/plugins/es_objects/include/graphene/es_objects/es_objects.hpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 Cryptonomex, Inc., and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+
+#include <graphene/app/plugin.hpp>
+#include <graphene/chain/database.hpp>
+
+namespace graphene { namespace es_objects {
+   using namespace chain;
+   //using namespace graphene::db;
+   //using boost::multi_index_container;
+   //using namespace boost::multi_index;
+
+//
+// Plugins should #define their SPACE_ID's so plugins with
+// conflicting SPACE_ID assignments can be compiled into the
+// same binary (by simply re-assigning some of the conflicting #defined
+// SPACE_ID's in a build script).
+//
+// Assignment of SPACE_ID's cannot be done at run-time because
+// various template automagic depends on them being known at compile
+// time.
+//
+#ifndef PROPOSAL_SPACE_ID
+#define PROPOSAL_SPACE_ID 6
```

Comment:
Better use a better name here.
Also I'm not sure if the number `6` is appropriate.

Reply:
this is something i dont understand fully so maybe you can explain me a bit. the comment states that plugins with the same ID will be added to the same binary so i think maybe is appropriate to have the same ID in the 2 ES plugins as: https://github.com/bitshares/bitshares-core/blob/develop/libraries/plugins/elasticsearch/include/graphene/elasticsearch/elasticsearch_plugin.hpp#L47

but maybe i am wrong, the snapshot plugin from @pmconrad  actually dont define a number(https://github.com/bitshares/bitshares-core/blob/develop/libraries/plugins/snapshot/include/graphene/snapshot/snapshot.hpp) please let me know if we should change this pulgin to number 7. 

in regards to the better name i am not sure what do you mean, there is no name defined here but only the ID. let me know. thanks @abitmore !

Reply:
I was referring that you use `es_object` for the plugin name but setting `PROPOSAL` here, it's not good practice. Not about the number.
According to the comment above, there should be a script to change the ID's at building time, but I'm not sure if there is one.
If I understood correctly, the space ID's are used to define a "space" for data storing in the object database. So, if no new object type is defined and need to be stored into object database, no need to set a `XX_SPACE_ID`, as have done with the snapshot plugin.

Reply:
oh, sorry about that name, didn't saw it.  it must be there because in a first version i was only going to store proposals. anyways, with your clarification i now removed all that section from the hpp file in commit https://github.com/bitshares/bitshares-core/pull/500/commits/7b453c6a05e4b052b6b04a304e094086b6b2a498 and code stills compiles and works fine.
thanks.